### PR TITLE
add web modeler configuration draft

### LINKED ISSUE #4330: Document the new SM web modeler deployment options
Relates to https://github.com/camunda/web-modeler/issues/10569

In SM, deploying from web modeler currently involves entering your cluster details in a form as part of the deployment step. This form can only save the details from one cluster at a time.

#10569 introduces the option to configure clusters via environment variables. When this configuration is present, the deployment dialog is updated to match that of web modeler - providing a list of configured clusters for the user to select from.

The documentation should be updated to:
- Add a list of the environment variables/configuration options to the Web Modeler [configuration docs](https://docs.camunda.io/docs/next/self-managed/modeler/web-modeler/configuration/#configuration-of-the-restapi-component)
- Ensure any references to the configuration in the docs are updated:
  - [Play docs](https://docs.camunda.io/docs/next/components/modeler/web-modeler/play-your-process/#use-play-with-camunda-self-managed)
  - [run a process](https://docs.camunda.io/docs/next/components/modeler/web-modeler/run-or-publish-your-process/)
  - [deploy your application](https://docs.camunda.io/docs/next/components/modeler/web-modeler/deploy-process-application/)
- Include any info in announcements/release notes/other locations 

Other notes: 
- The Camunda docker compose installation (used as an example in the Play docs and possibly other locations) provides a pre-configured cluster; docs may need to be updated to note this.

Configuration options (details in https://github.com/camunda/web-modeler/issues/10569)
```
CAMUNDA_MODELER_CLUSTERS_0_NAME=test cluster 1
CAMUNDA_MODELER_CLUSTERS_0_VERSION=8.6.0
CAMUNDA_MODELER_CLUSTERS_0_AUTHENTICATION=OAUTH
CAMUNDA_MODELER_CLUSTERS_0_URL_ZEEBE_GRPC=https://zeebe-1.example.com:26500
CAMUNDA_MODELER_CLUSTERS_0_URL_ZEEBE_REST=https://zeebe-1.example.com:8080
CAMUNDA_MODELER_CLUSTERS_0_URL_OPERATE=https://operate-1.example.com
CAMUNDA_MODELER_CLUSTERS_0_URL_TASKLIST=https://tasklist-1.example.com
CAMUNDA_MODELER_CLUSTERS_0_OAUTH_URL=https://auth.example.com/token
CAMUNDA_MODELER_CLUSTERS_0_OAUTH_SCOPE=test-scope
CAMUNDA_MODELER_CLUSTERS_0_OAUTH_AUDIENCE_ZEEBE=test-zeebe
CAMUNDA_MODELER_CLUSTERS_0_OAUTH_AUDIENCE_OPERATE=test-operate
CAMUNDA_MODELER_CLUSTERS_0_OAUTH_AUDIENCE_TASKLIST=test-tasklist

CAMUNDA_MODELER_CLUSTERS_1_NAME=test cluster 2
CAMUNDA_MODELER_CLUSTERS_1_VERSION=8.7.0-alpha1
CAMUNDA_MODELER_CLUSTERS_1_URL_ZEEBE_GRPC=https://zeebe-2.example.com:26500
CAMUNDA_MODELER_CLUSTERS_1_URL_ZEEBE_REST=https://zeebe-2.example.com:8080
CAMUNDA_MODELER_CLUSTERS_1_URL_OPERATE=https://operate-2.example.com
CAMUNDA_MODELER_CLUSTERS_1_URL_TASKLIST=https://tasklist-2.example.com
CAMUNDA_MODELER_CLUSTERS_1_OAUTH_URL=https://auth.example.com/token
```

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Closes #4330 

Adds the web modeler cluster configuration options to the SM docs, and updates existing pages to link to the configuration options, and note the differences in deploying when SM installations have no provided config.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [X] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/modeler/web-modeler/configuration/configuration.md [hunk 1]
```diff
@@ -107,6 +107,27 @@ Refer to the [advanced logging configuration guide](./logging.md#logging-configu
 
 Refer to the [advanced SSL configuration guide](./ssl.md) for additional details on how to set up secure connections (incoming & outgoing) to the Web Modeler components.
 
+### Clusters
```
Comment (nithinssabu) [inline docs/self-managed/modeler/web-modeler/configuration/configuration.md:110]: 🔧 I think the `Clusters` sub-section can be moved to the top and be the first one under the `Configuration of the restapi component` section as it is an important configuration.

File: docs/self-managed/modeler/web-modeler/configuration/configuration.md [hunk 2]
```diff
@@ -107,6 +107,27 @@ Refer to the [advanced logging configuration guide](./logging.md#logging-configu
...
+
+To add additional clusters, increment the `0` value for each variable (`CAMUNDA_MODELER_CLUSTERS_1_NAME`).
+
+| Environment variable                                 | Description                                                    | Example value                       |
+| ---------------------------------------------------- | -------------------------------------------------------------- | ----------------------------------- |
+| `CAMUNDA_MODELER_CLUSTERS_0_NAME`                    | The name of your cluster.                                      | `test cluster 1`                    |
+| `CAMUNDA_MODELER_CLUSTERS_0_VERSION`                 | The Camunda version used by this cluster.                      | `8.6.0`                             |
+| `CAMUNDA_MODELER_CLUSTERS_0_AUTHENTICATION`          | The authentication to use with your cluster.                   | `OAUTH`, `NONE`                     |
+| `CAMUNDA_MODELER_CLUSTERS_0_URL_ZEEBE_GRPC`          | The address where your cluster can be reached.                 | `https://zeebe-1.example.com:26500` |
```
Comment (nithinssabu) [inline docs/self-managed/modeler/web-modeler/configuration/configuration.md:121]: ❌ The example value for grpc url can be changed to `grpcs://zeebe-1.example.com:26500`

**Review by nithinssabu — COMMENTED** [review]
Great job @conceptualshark 🚀 Thanks for the contribution 🙂 

Please have a look at the suggestions below:

File: docs/self-managed/modeler/web-modeler/configuration/configuration.md (hunk 1 above)
Comment (nithinssabu) [reply docs/self-managed/modeler/web-modeler/configuration/configuration.md:110]: It is also worth mentioning that in docker installations, the local zeebe cluster will be pre-configured. Starting with 8.7, we will also provide the pre-configuration in helm installations.

Comment (conceptualshark) [reply docs/self-managed/modeler/web-modeler/configuration/configuration.md]: 👍 Moved this section to the top of the `restapi` section, and added the Docker default. I can add the 8.7 Helm option when that becomes available.

File: docs/self-managed/modeler/web-modeler/configuration/configuration.md (hunk 2 above)
Comment (conceptualshark) [reply docs/self-managed/modeler/web-modeler/configuration/configuration.md]: 👍 Updated!

File: docs/components/modeler/web-modeler/play-your-process.md [hunk 3]
```diff
@@ -14,11 +14,9 @@ Play is a Zeebe-powered playground environment within Web Modeler for validating
 
 To use Play, open a BPMN diagram and click the **Play** tab. Read the [limitations and availability section](#limitations-and-availability) if this section is missing.
 
-In Self-Managed, you will be prompted to provide the [details](#use-play-with-camunda-self-managed) of your cluster:
+In Self-Managed, you are prompted to select from any clusters defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). The Camunda 8 Docker Compose distribution provides one cluster configured by default. If no configuration is found, you are prompted to enter your cluster details [manually](#use-play-with-camunda-self-managed).
```
Comment (mesellings) [inline docs/components/modeler/web-modeler/play-your-process.md:17]:
```suggestion
In Self-Managed, you are prompted to select from the clusters defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). The Camunda 8 Docker Compose distribution provides one cluster configured by default. If no configuration is found, you are prompted to [manually enter your cluster details](#use-play-with-camunda-self-managed).
```
Minor suggestions

File: docs/components/modeler/web-modeler/play-your-process.md [hunk 4]
```diff
@@ -14,11 +14,9 @@ Play is a Zeebe-powered playground environment within Web Modeler for validating
...
 
-![play cluster config](img/play-cluster-configuration.png)
-
-This starts a Play environment that utilizes your selected development cluster in SaaS, or the specified cluster in a Self-Managed setup.
+A Play environment is then started that utilizes your selected development cluster in SaaS, or the specified cluster in a Self-Managed setup.
```
Comment (mesellings) [inline docs/components/modeler/web-modeler/play-your-process.md:19]:
```suggestion
A Play environment then starts, using your selected development cluster in SaaS, or the specified cluster in a Self-Managed setup.
```
Suggestion, feel fee to ignore

File: docs/components/modeler/web-modeler/play-your-process.md [hunk 5]
```diff
@@ -179,9 +177,9 @@ Prior to the 8.6 release, Play can be accessed by installing the 8.6.0-alpha [He
 
 ## Use Play with Camunda Self-Managed
 
-After selecting the **Play** tab in Self-Managed, you are prompted to provide the details of your cluster.
+After selecting the **Play** tab in Self-Managed, you are prompted to select from any clusters defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). The Camunda 8 Docker Compose distribution provides one cluster configured by default.
```
Comment (mesellings) [inline docs/components/modeler/web-modeler/play-your-process.md:180]:
```suggestion
After selecting the **Play** tab in Self-Managed, you are prompted to select from the clusters defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). The Camunda 8 Docker Compose distribution provides one cluster configured by default.
```
Suggestion

File: docs/components/modeler/web-modeler/play-your-process.md [hunk 6]
```diff
@@ -179,9 +177,9 @@ Prior to the 8.6 release, Play can be accessed by installing the 8.6.0-alpha [He
...
 
-See the table below for the requirement for each field, as well as an example value:
+If no cluster is configured, Web Modeler will request the following cluster details to use for deployment:
```
Comment (mesellings) [inline docs/components/modeler/web-modeler/play-your-process.md:182]:
```suggestion
If no cluster is configured, Web Modeler requests the following cluster details to use for deployment:
```

File: docs/components/modeler/web-modeler/run-or-publish-your-process.md [hunk 7]
```diff
@@ -37,6 +37,8 @@ To deploy, click **Deploy** in the upper right corner of the modeling screen:
 
 ![The deploy dialog of a BPMN diagram](img/web-modeler-deploy.png)
 
+In Self-Managed, you can deploy your diagram to any cluster defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). If no configuration is found, you are prompted to enter your cluster details manually.
```
Comment (mesellings) [inline docs/components/modeler/web-modeler/run-or-publish-your-process.md:40]:
```suggestion
In Self-Managed, you can deploy your diagram to any cluster defined in your Web Modeler [configuration](/self-managed/modeler/web-modeler/configuration/configuration.md#clusters). If no configuration is found, you are prompted to manually enter your cluster details.
```

**Review by mesellings — APPROVED** [review]
@conceptualshark Great writing, only a few very minor  (non-blocking) suggestions you can ignore or implement as you see fit 👍 🥇

**Review by nithinssabu — DISMISSED** [review]

**Comment by conceptualshark** [comment]
Thank you both for the reviews! @mesellings, I've implemented your suggestions, if you don't mind giving this a final pass.

(Not sure why the merge from main removed an unrelated line of whitespace, but I figure it isn't the end of the world.)

**Review by mesellings — APPROVED** [review]
Lgtm! 🚀

**Pull request merged**